File: 백준/Silver/3986. 좋은 단어/좋은 단어.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int N = Integer.parseInt(br.readLine());
		
		int result = 0;
		for (int t = 0; t < N; t++) {
			char[] word = br.readLine().toCharArray();
			Stack<Character> stack = new Stack<>();

			for (char w : word) {
				if (!stack.isEmpty() && w == stack.peek()) {
					stack.pop();
				} else {
					stack.push(w);
				}
			}
			
			if (stack.isEmpty()) {
				result++;
			}
		}
		System.out.println(result);
	}
}

File: 백준/Silver/3986. 좋은 단어/좋은 단어.py
from sys import stdin
input = stdin.readline

def check(word):
    stack = []
    for w in word:
        if stack:
            if stack[-1] == w:
                stack.pop()
            else:
                stack.append(w)
        else:
            stack.append(w)
    return 0 if stack else 1

def solve():
    N = int(input())

    result = 0
    for _ in range(N):
        word = input().strip()
        if check(word):
            result += 1
    print(result)

if __name__ == "__main__":
    solve()